test/test_support.ts

```diff
@@ -24,45 +24,46 @@ const {cachedLibPath, cachedLib} = (function() {
   return {cachedLibPath: p, cachedLib: host.getSourceFile(fn, ts.ScriptTarget.ES6)};
 })();
 
-export function annotateSource(src: string, options: SickleOptions = {}): string {
+export function annotateSource(inputFileName: string, sourceText: string, options: SickleOptions = {}): string {
   var host = ts.createCompilerHost(OPTIONS);
   var original = host.getSourceFile.bind(host);
   host.getSourceFile = function(
                            fileName: string, languageVersion: ts.ScriptTarget,
                            onError?: (msg: string) => void): ts.SourceFile {
     if (fileName === cachedLibPath) return cachedLib;
-    if (fileName === 'main.ts') {
-      return ts.createSourceFile(fileName, src, ts.ScriptTarget.Latest, true);
+    if (fileName === inputFileName) {
+      return ts.createSourceFile(fileName, sourceText, ts.ScriptTarget.Latest, true);
     }
     return original(fileName, languageVersion, onError);
   };
 
-  var program = ts.createProgram(['main.ts'], OPTIONS, host);
-  if (program.getSyntacticDiagnostics().length) {
-    throw new Error(formatDiagnostics(ts.getPreEmitDiagnostics(program)));
```

Comment:
I wrote a little summary to typescript-eng, search your mail for [getPreEmitDiagnostics] for a summary!